cmd/kube-apiserver/app/server.go

```diff
@@ -120,6 +121,9 @@ cluster's shared state through which all other components interact.`,
 
 	fs := cmd.Flags()
 	namedFlagSets := s.Flags()
+	verflag.AddFlags(namedFlagSets.FlagSet("global"))
+	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
+	options.AddGenericFlags(namedFlagSets.FlagSet("generic"))
 	for _, f := range namedFlagSets.FlagSets {
```

Comment:
IMO, add the generic like that, no need to add function
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L270

Reply:
We can do that for `help` because we know the type of it. We do not know the types for flags like `cloud-provider-gce-lb-src-cidrs`, `default-not-ready-toleration-seconds`, etc. and one way to obtain the type is to extract it out from `flag.CommandLine` using the `Register` function in `globalflag` (It does a lookup, normalizes the pflag, and add it to the provided flagset). What do you think?

Reply:
sounds good :+1:

Reply:
Updated. What do you think of the new approach?

Reply:
look better now